### DISCUSSION TITLE
The C:\Windows\SystemTemp Becomes Full and Is Not Cleared

### DISCUSSION THREAD

**Discussion by enasbond** [fo-dicom/fo-dicom#1830] [Q&A]
I have an SCP store application, and I encountered the following exception: "Exception processing P-Data-TF PDU - System.IO.IOException: There is not enough space on the disk. : 'C:\Windows\SystemTemp\apdymv40.h0y'." The C: drive lacks sufficient space.

The main issue is that this folder becomes full and is not cleared, which affects the system due to insufficient disk space on the C: drive.

Is there a way to force clear this folder or configure it to use another storage location?

Environment

Fellow Oak DICOM version: 5.1.3
OS: Windows Server 2022
.NET Core 8

**Comment by mrbean-bremen** [comment]
You can [set the used temp file yourself](https://fo-dicom.github.io/stable/v5/api/FellowOakDicom.IO.TemporaryFile.html#FellowOakDicom_IO_TemporaryFile_StoragePath):
```cs
TemporaryFile.StoragePath = your_temp_path;
```
This is a static method, so you should call it once before you create the server. If it is not set, it defaults to the system temp path.

**Comment by enasbond** [comment]
Thank you so much for your quick response.

My main issue is that the files are not being removed from the temp folder. Over time, with each association received, the files accumulate in the folder until it becomes full, causing the system to stop working. I need to understand why the files are not being removed from the temp folder. Is there anything specific I need to do with the DICOM service to handle this issue?

**Comment by mrbean-bremen** [comment]
This sounds somewhat like #557. I'm actually not sure what the state of this is...
